kart/spatial_tree.py

```diff
@@ -292,6 +365,11 @@ def update_spatial_tree(repo, commit_spec, verbosity=1, clear_existing=False):
                 params,
             )
 
+        # Update indexed commits.
+        params = [(bytes.fromhex(commit_id),) for commit_id in all_independent_commits]
+        dbcur.execute("DELETE FROM commits;")
```

Comment:
there's a race condition here, if two of these things run in parallel.

can/should/do we prevent them from running in parallel? Perhaps by just taking an exclusive lock on the db before we fetch the commit IDs up above. (but that'd mean we'd have to use the same connection/session throughout)

Reply:
Talked about this offline - since we're using WAL, a single writer gets to hold the lock on the DB and update in a self-consistent way. Since the reads are done separately to the write, there's a possibility of two different writers both trying to re-index the same commits, but since that's supposed to be idempotent (see the tests) that's just wasted effort - it won't actually lead to any inconsistency. WDYT?